elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -0,0 +1,276 @@
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Data
+/**
+ * Util class to validate the model config files in local before deployment.
+ */
+public class DynamicConfigValidator {
+
+    private static final Set<String> SQL_DISALLOWED_WORDS = new HashSet<>(
+            Arrays.asList("DROP", "TRUNCATE", "DELETE", "INSERT", "UPDATE", "ALTER", "COMMENT", "CREATE", "DESCRIBE",
+                    "SHOW", "USE", "GRANT", "REVOKE", "CONNECT", "LOCK", "EXPLAIN", "CALL", "MERGE", "RENAME"));
+    private static final String[] ROLE_NAME_DISALLOWED_WORDS = new String[] { "," };
+    private static final String SQL_SPLIT_REGEX = "\\s+";
+    private static final String SEMI_COLON = ";";
+    private static final Pattern HANDLEBAR_REGEX = Pattern.compile("<%(.*?)%>");
+    private static final Options OPTIONS = prepareOptions();
+
+    private ElideTableConfig elideTableConfig;
+    private ElideSecurityConfig elideSecurityConfig;
+    private Map<String, Object> variables;
+    private String configDir;
+
+    public static void main(String[] args) throws IOException, ParseException {
+
+        CommandLine cli = new DefaultParser().parse(OPTIONS, args);
+
+        if (cli.hasOption("help")) {
+            printHelp();
+            return;
+        }
+        if (!cli.hasOption("configDir")) {
+            printHelp();
+            throw new MissingOptionException("Missing required option");
+        }
+        String configDir = cli.getOptionValue("configDir");
+        File file = new File(configDir);
+        String absoluteBasePath = DynamicConfigHelpers.formatFilePath(file.getAbsolutePath());
+        log.info("Absolute Path for Model Configs Directory: " + absoluteBasePath);
+
+        if (!file.isDirectory()) {
+            throw new IllegalStateException("Model Configs Directory doesn't exists");
+        }
+
+        DynamicConfigValidator dynamicConfigValidator = new DynamicConfigValidator();
+        dynamicConfigValidator.setConfigDir(absoluteBasePath);
+
+        dynamicConfigValidator.readVariableConfig();
+        if (dynamicConfigValidator.readSecurityConfig()) {
+            validateRoleInSecurityConfig(dynamicConfigValidator.getElideSecurityConfig());
+        }
+        if (dynamicConfigValidator.readTableConfig()) {
+            validateSqlInTableConfig(dynamicConfigValidator.getElideTableConfig());
+        }
+
+        log.info("Configs Validation Passed!");
+    }
+
+    /**
+     * Read variable file config.
+     * @return boolean true if variable config file exists else false
+     * @throws JsonProcessingException
+     */
+    private boolean readVariableConfig() throws JsonProcessingException {
```

Comment:
Does the dynamic logic read the files twice?  Once to build the config for compilation and once to validate?  We really ought to read the config once to make sure we are validating the same configuration we are running with.

Reply:
The main function however can separate read and validate - but the validator needs to be hooked into two places - the utility main function and the normal path when the service launches and compiles the configs.

Reply:
@aklish  @moizarafat 
> Does the dynamic logic read the files twice? Once to build the config for compilation and once to validate? We really ought to read the config once to make sure we are validating the same configuration we are running with.

Splitted getElideTablePojo & getElideSecurityPojo methods of DynamicConfigHelpers class such that we read the configs only once.

> The main function however can separate read and validate - but the validator needs to be hooked into two places - the utility main function and the normal path when the service launches and compiles the configs.

Created a separate class method to read and validate config files which can be called outside of Main function also. Please check if this is what we wanted.